packages/openspp/src/Adaptor.js

```diff
@@ -660,6 +658,311 @@ export function unenroll(registrant_id, program_id) {
   };
 }
 
+/**
+ * create new individual for OpenSPP
+ * @public
+ * @example
+ * createIndividual({ name: "Individual 1" })
+ * @function
+ * @param {object} data - registrant create data
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function createIndividual(data, callback=false) {
+  return async state => {
+    try {
+      if (typeof data !== 'object' || Array.isArray(data) || data === null) {
+        throw new Error(`${data} is not a create object!`);
+      }
+      if (!data.name) {
+        throw new Error(`"name" is a required parameter!`);
+      }
+      data.is_registrant = true;
+      data.is_group = false;
+      let individualId = await sppConnector.create('res.partner', data);
+      let res = await sppConnector.searchRead(
+        'res.partner',
+        [['id', '=', individualId]],
+        ['registrant_id'],
+        { limit: 1 }
+      );
+      let individualRegistrantId = res[0].registrant_id;
+      console.log(`ℹ Individual created with registrant ID: ${individualRegistrantId}`);
+      let nextState = composeNextState(state, individualRegistrantId);
+      if (callback) {
+        return callback(nextState);
+      }
+      return nextState;
+    } catch (err) {
+      console.log(`✗ Error: ${err}`);
```

Comment:
I think I gave you bad advice before - I'm not sure what I was thinking now! 

You may not always want to return state if there's an error. It depends on the error really. Sometimes it's fine to throw the error and let the job "crash". You should only trap the error if it's non-fatal or the next operation should still run.

In this case, if there's a validation error on the data, the job will carry on running. That might not be what you want - it might be better for the job to die, and alert the team that action needs to be taken.